include/mixer/mixer.hh

```diff
@@ -71,15 +74,11 @@ template <std::size_t num_banks> class Mixer {
                 }
                 output_pipe.cond.notify_all();
             }
+            return true;
         });
     }
 
-    void stop() {
-        run_thread = false;
-    }
-
     ~Mixer() {
-        run_thread = false;
         thread.join();
```

Comment:
What happens here if you instantiate this class but don't start a thread. I've found with the RMS that this causes issues. I can see you changed your tests to all have a thread running, which works I guess.

Reply:
I was actually thinking about that, is there any benefit to having a `run()` method when I could do the same thing in the constructor?

Reply:
who knows

Reply:
can you have an opinion please?

Reply:
Anyway, could mixer's destructor be called without a running thread. The thread is private right? So it would only be the case that maybe something went wrong and you want to destruct it without the thread running.

Reply:
The mixer's destructor could be called without running a thread, which is why other classes have the `thread_alive` variable which is checked in the destructor. However, the way I have designed the mixer there is no reason to have an object without a thread running, since the only other method is static so can be tested without the object existing. 

Probably not worth changing at this point? Better to make progress now and come back and review it later if we have time.